### PR TITLE
allow custom repo inside tags

### PR DESCRIPTION
**Pull request by cescoferraro**
solves #69

### CODE REVIEW

File: tasks/image/tag.go [hunk 1]
```diff
@@ -24,6 +25,14 @@ func tagImage(ctx *context.ExecuteContext, t *Task, imageTag string) error {
 		return nil
 	}
 
+	switch strings.Count(imageTag, ":") {
+	case 1:
+	case 2:
+		imageTag = imageTag[strings.Index(imageTag, ":")+1:]
+	default:
+		return fmt.Errorf("Not a valid tag")
```
Comment (dnephin) [inline tasks/image/tag.go]: I think we need `case 0` to support the old behaviour

File: tasks/image/tag.go [hunk 2]
```diff
@@ -2,8 +2,9 @@ package image
 
 import (
 	"fmt"
-
 	"github.com/dnephin/dobi/tasks/context"
+	"strings"
+
```
Comment (dnephin) [inline tasks/image/tag.go]: The sorting here is wrong. `strings` should be with `fmt` as a "build-in" import

File: tasks/image/tag.go (hunk 1 above)
Comment (cescoferraro) [inline tasks/image/tag.go]: needs case 1 to figure out whenever its a repo or a tag depending on the existence of a slash

Comment (dnephin) [inline tasks/image/tag.go]: Yes, it needs case 1 and 2, but I think it also needs `case 0`

**Comment by dnephin** [comment]
I looked into this further and I realized that `GetImageName()` also needs to change. If the first tag is not just a `:tag` it's going to create a bad name.

**Comment by dnephin** [comment]
And the check in `config/` is at `(c *ImageConfig) ValidateImage()`. That needs to change.

**Comment by dnephin** [comment]
Tests are passing, so going to merge.

Thanks!

**Pull request merged**